File: test-plugins/org.yakindu.sct.generator.cpp.test/test-gen/org/yakindu/sct/generator/cpp/test/InEventLifeCycleTest.java
/**
* Copyright (c) 2013 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
	package org.yakindu.sct.generator.cpp.test;
		
	import java.util.Collection;
	
	import org.junit.Before;
	import org.junit.runner.RunWith;
	import org.yakindu.sct.generator.c.gtest.GTest;
	import org.yakindu.sct.generator.c.gtest.GTestRunner;
	import org.yakindu.sct.generator.c.gtest.GTestHelper;
	
	@GTest(sourceFile = "gtests/InEventLifeCycleTest/InEventLifeCycleTest.cc", program = "gtests/InEventLifeCycleTest/InEventLifeCycle", model = "testmodels/SCTUnit/InEventLifeCycle.sct" )
	@RunWith(GTestRunner.class)
	public class InEventLifeCycleTest {
	
		protected final GTestHelper helper = new GTestHelper(this) {
	
		@Override
		protected void getSourceFiles(Collection<String> files) {
			super.getSourceFiles(files);
		files.add(getFileName(getTestProgram()) + ".cpp");
		}
		};
	
		@Before
		public void setUp() {
		helper.generate();
		helper.compile();
		}
	}

File: test-plugins/org.yakindu.sct.generator.c.test/gtests/InEventLifeCycle/InEventLifeCycleTest.cc
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include "gtest/gtest.h"
#include "InEventLifeCycle.h"

static InEventLifeCycle statechart;

class StatemachineTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		inEventLifeCycle_init(&statechart);
	}
};


TEST_F(StatemachineTest, eventLifeCycle) {					
	inEventLifeCycle_enter(&statechart);
	inEventLifeCycleIface_raise_e(&statechart);
	EXPECT_TRUE(inEventLifeCycleIface_get_i(&statechart)== 0l);
	inEventLifeCycle_runCycle(&statechart);
	EXPECT_TRUE(inEventLifeCycleIface_get_i(&statechart)== 1l);
}


